public/API/API_GetClaims.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Validation\Rule;
+use LegacyApp\Community\Enums\ClaimFilters;
+
+/*
+ *  API_GetClaims - returns information about 1000 max set claims.
+ *    k : claim kind - 1 for completed, 2 for dropped, 3 for expired (default: 1)
+ *
+ *  array
+ *   object     [value]
+ *    string     ID                 unique ID of the claim
+ *    string     User               user who made the claim
+ *    int        UserRole           role of the user who made the claim
+ *    string     GameID             id of the claimed game
+ *    string     GameTitle          title of the claimed game
+ *    string     GameIcon           site-relative path to the game's icon image
+ *    string     ConsoleName        console name of the claimed game
+ *    string     ClaimType          claim type: 0 - primary, 1 - collaboration
+ *    string     SetType            set type claimed: 0 - new set, 1 - revision
+ *    string     Status             claim status: 0 - active, 1 - complete, 2 - dropped
+ *    string     Extension          number of thes the claim has been extended
+ *    string     Special            flag indicating a special type of claim
+ *    string     Created            date the claim was made
+ *    string     DoneTime           date the claim is done
+ *                                    Expiration date for active claims
+ *                                    Completion date for complete claims
+ *                                    Dropped date for dropped claims
+ *    string     Updated            date the claim was updated
+ *    string     MinutesLeft        time in minutes left until the claim expires
+ */
+
+ $input = Validator::validate(Arr::wrap(request()->query()), [
+    'k' => [
+        'nullable',
+        Rule::in(['1', '2', '3']),
+    ],
+], [
+    'k.in' => 'k must be set to one of the following values: :values',
+]);
+
+$completedClaims = '1';
+$droppedClaims = '2';
+$expiredClaims = '3';
+
+$claimKind = request()->query('k', $completedClaims);
+
+$claimFilter = ClaimFilters::AllCompletedPrimaryClaims;
+if ($claimKind === $droppedClaims) {
+    $claimFilter = ClaimFilters::AllDroppedClaims;
+} elseif ($claimKind === $expiredClaims) {
+    $claimFilter = ClaimFilters::AllActiveClaims;
```

Comment:
Would it be beneficial to have an `$activeClaims='4'` to return the unfiltered active claims, or is that too redundant with `API_GetActiveClaims`?

Reply:
> I would expect "expired" claims to only return active claims past their Finished date. It would probably be easiest to do that as a post-processing step here (`MinutesLeft < 0`), but `getFilteredClaims` has a `getExpiringOnly` flag that could be expanded to an enum and then the filtering could be done at the query level.

Good point. I see where we would do this in the query itself. It appears `getFilteredClaims()` has several consumers. 
To reduce risk, I've opted for the post-processing approach in the API endpoint itself (https://github.com/RetroAchievements/RAWeb/commit/b3e44d98130302069814d21d7e7cdc790c808ab0).

> Would it be beneficial to have an `$activeClaims='4'` to return the unfiltered active claims, or is that too redundant with `API_GetActiveClaims`?

I considered the same thing and decided it was redundant and should not be added. I am happy to defer to your judgment here - if you believe we should have a `4` argument for active, I will add it.

Reply:
Sounds good. I will not add a `4` option and will leave the existing endpoint alone. Agreed, we can always add it later if there's any demand for it.